todo/material/task.md

```diff
@@ -0,0 +1,110 @@
+# Aufgabenliste
+
+::: info
+Diese Aufgabe basiert auf _keinem CS50 Practice Problem_.
+
+**Disclaimer:** Diese Aufgabe wurde nicht vom Lehrstuhl herausgegeben und kann Fehler enthalten. Sie dient, wie das gesamte Material von [inf-lab.dev](https://inf-lab.dev), lediglich zu Übungszwecken!
+:::
+
+Wir alle haben verschiedene Aufgaben, jedoch ist es nicht immer einfach sich alle seine Aufgaben zu merken. Daher wollen wir in dieser Aufgabe eine ToDo-Liste entwickeln, welche uns erlaubt unsere Aufgaben über die Konsole zu verwalten.
+
+## Teilaufgabe 1
+
+Wie immer, wenn wir einen vorbereiteten Code bekommen, sollten wir diesen vorerst verstehen bevor wir Änderungen vornehmen können. Lies deshalb den Code in der Datei `todo.py` und beantworte insbesondere folgende Fragen. Wenn du noch etwas unsicher beim Python Syntax bist, kannst du die vorgegebene Datei `todo.c` zur hilfe nehmen. Diese enthält den selben Code - nur für die Programmiersprache C.
+
+-   Welche Funktionen müssen laut `TODO`-Kommentaren von uns implementiert werden? Was sind ihre Parameter?
+-   Wie genau kann der Benutzer eine Aufgabe hinzufügen, nachdem er das Programm mit `python todo.py` gestartet hat?
+-   Welche Aufgabe haben die `NotImplementedError`s, die bei verschiedenen Funktionen ausgelöst werden?
+
+## Teilaufgabe 2
+
+Nachdem wir nun das Codegerüst verstanden haben, möchten wir in dieser Aufgabe den Grundstein für das Programm legen.
+
+Erstelle daher in der bereitgestellten Datei `todo.py` eine Klasse `ToDo`, welche die eigentliche Aufgabe (`task`) und ob diese abgeschlossen ist (`completed`), speichert.
+Diese Klasse soll mindestens folgende Methoden besitzen:
+
+-   `__init__`: Initialisiert die Aufgabe und speichert den übergebenen `task`. Inital ist eine Aufgabe immer nicht `completed`.
+-   `complete`: Markiert die Aufgabe als abgeschlossen (`completed`).
+-   `update`: Ersetzt den gespeicherten `task` mit einem neuen, übergebenen `new_task`.
+
+Füge außerdem eine globale Variable `todos` hinzu, in welche die Liste aller `ToDo`s gespeichert wird.
+
+Implementiere schließlich die Funktion `show`, welche die Liste aller Aufgaben im untenstehenden Format ausgibt. Die Zahl in den eckigen Klammern `[]`, ist hierbei der Index (die Position) der Aufgabe innerhalb der `todos` Liste. Der Text `(completed)` soll außerdem nur angezeigt werden, wenn die Aufgabe auch wirklich abgeschlossen ist.
+
+```
+[0] Bake a cake (completed)
+[1] Learn Inf-Einf
+[2] Wash my clothes
+```
+
+Sollte noch keine Aufgabe existieren, soll statt der obigen Ausgabe die Nachricht `No ToDos have been created yet.` ausgegeben werden.
+
+## Teilaufgabe 3
+
+Da die Grundfunktionalität des Programms nun funktioniert, ist es an der Zeit auch die verbleibenden Funktionen zu implementieren.
```

Comment:
"Da die Grundfunktionalität des Programms nun gegen ist, ..." <- That's what I would prefer.